### ISSUE TITLE
Create AWS Cloudformation template for creating Teleport. 

### ISSUE THREAD

**Issue by benarent** [gravitational/teleport#2806]
As I work on AWS docs and Guides, I've been following some AWS Marketplace best practices and it's now possible to quickly start a Teleport Cluster using Cloudformation. 

https://docs.aws.amazon.com/marketplace/latest/userguide/cloudformation.html 
https://aws.amazon.com/blogs/awsmarketplace/cloudformation-templates-101-for-sellers-in-aws-marketplace/

See benarent/docs/aws-guide branch for AWS docs in the works.

**Comment by webvictim** [comment]
We do actually have a load of CloudFormation templates already:

VPC template: https://github.com/gravitational/teleport/blob/master/assets/marketplace/cloudformation/vpc.yaml
OSS: https://github.com/gravitational/teleport/blob/master/assets/marketplace/cloudformation/oss.yaml
Enterprise: https://github.com/gravitational/teleport/blob/master/assets/marketplace/cloudformation/ent.yaml

I split them out to make it easier to launch into an existing VPC if people wanted to. We used to auto-build AMIs and test the CloudFormation via Jenkins too but it wasn't very reliable, so I disabled it.

This hasn't been updated in a while (since around Teleport 2.7.0) but this is probably a good place to start.

**Comment by benarent** [comment]
Awesome, Thanks @webvictim. I now there are a bunch of other steps to productize the CloudFormation template for the marketplace but this is a great start.

**Comment by webvictim** [comment]
Yeah - we actually uploaded these CloudFormation templates to AWS Marketplace once before and started going through the listing process but decided after feedback from AWS themselves that it was too complicated.

Marketplace doesn't provide a way for you to get arbitrary inputs from the user and provide them to the templates, which means that the user has to really know what they're doing and for most people it was just far too involved. Our example Terraform is much nicer so we started using that instead.
